### ISSUE TITLE
Enhancement: Generic driver node cleanup after "docker-machine rm"

### ISSUE THREAD

**Issue by dayreiner** [docker/machine#3231]
When removing a machine provisioned with the generic driver, the node is removed from the machine list but the system itself is untouched. If you then wish to re-create the same generic host with machine, the provisioning fails (due to #2970 #2976 and others). It requires manual intervention to get the system back to a state where it can be created in machine again -- at least, that's been my experience so far with CentOS hosts using the generic driver.

Other drivers do things like cancel the instance when a machine is removed. It would be nice to have an _optional_ flag (maybe `--cleanup`) to tell the generic driver to also remove the various pieces it added with `docker-machine create`, bringing the system back to its initial state. Cleanup tasks may include:
- Stopping docker
- Uninstalling docker packages
- Removing any added yum/apt repos
- Removing the systemd override file (/etc/systemd/system/docker.service)
- Removing any leftover docker dirs (/var/lib/docker, /etc/docker)
- Cleaning up any snapshots under /var/lib/docker (i.e. if using btrfs storage driver)

... and perhaps other tasks, depending on the OS.

**Comment by nathanleclaire** [comment]
Hm, yeah, I think this kind of fits in the general category of "making provisioning idempotent"-- rather than implement tear-down logic, I think it would be easier in the short term to simply make it safe to call provisioning multiple times.

**Comment by dayreiner** [comment]
True, the main issue is really to prevent a re-provision from failing. Maybe a confirmation though to prevent accidents.
